pkgs/development/interpreters/elixir/generic-builder.nix

```diff
@@ -0,0 +1,77 @@
+{ pkgs, stdenv, fetchFromGitHub, erlang, rebar, makeWrapper,
+  coreutils, curl, bash, debugInfo ? false }:
+
+{ baseName ? "elixir"
+, version
+, minimumOTPVersion
+, sha256 ? null
+, rev ? "v${version}"
+, src ? fetchFromGitHub { inherit rev sha256; owner = "elixir-lang"; repo = "elixir"; }
+}:
+
+let
+  inherit (stdenv.lib) getVersion versionAtLeast;
+
+in
+  assert versionAtLeast (getVersion erlang) minimumOTPVersion;
```

Comment:
If you end up not using the Erlang-specific getVersion, then please delete it in this PR.

IMO the circular dependency is not a problem, since this is a lazy language - it is actually full of circular dependencies (at least at data level).

I think there might be a better solution - to make sure that `erlangR16.version` is actually `16.3.1` and hide the fact that Erlang/OTP had funny versioning.

Feel free to choose any of the discussed solutions, I do not think this is important enough to stall this PR any longer.